src/components/Footer/Footer.js

```diff
@@ -63,23 +85,104 @@ const Footer = ({
     }
     return COLORS.balance_gray;
   };
+
+  const username = item?.anon_user_info_emoji_name
+    ? `Anonymous ${item?.anon_user_info_emoji_name}`
+    : item?.actor?.data?.username;
+
+  const onPressDM = async () => {
+    try {
+      setLoading({...loading, loadingDm: true});
+      if (!item?.anon_user_info_emoji_name) {
+        const channelName = [username, profile?.myProfile?.username].join(',');
+        const selectedUser = {
+          user: {
+            name: channelName,
+            image: item?.actor?.data?.profile_pic_url || DEFAULT_PROFILE_PIC_PATH
+          }
+        };
+        const members = [item?.actor?.id, profile.myProfile.user_id];
+        await createSignChat(members, selectedUser);
+      } else {
+        await sendMessageDM(item.id, 'post', 'SIGNED');
+      }
+    } catch (e) {
+      console.warn(e);
+    } finally {
+      refSheet.current.close();
+      setLoading({...loading, loadingDm: false});
+    }
+  };
+
+  const onPressDMAnon = async () => {
+    try {
+      setLoading({...loading, loadingDmAnon: true});
+      await sendMessageDM(item.id, 'post', 'ANONYMOUS');
+    } catch (e) {
+      console.warn(e);
+    } finally {
+      refSheet.current.close();
+      setLoading({...loading, loadingDmAnon: false});
+    }
+  };
+
+  const dataSheet = [
+    {
+      id: 1,
+      name: loading.loadingDm ? 'Loading...' : `Message ${username}`,
+      icon: <SendDMBlack />,
+      onPress: onPressDM
+    },
+    {
+      id: 2,
+      name: loading.loadingDmAnon ? 'Loading...' : `Message ${username} anonymously`,
+      icon: <SendDMAnonBlock />,
+      onPress: onPressDMAnon
+    },
+    {
+      id: 3,
+      name: 'Share link',
+      icon: <ShareBlack />,
+      onPress: onPressShare
+    }
+  ];
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `onPressDM` and `onPressDMAnon` functions are making asynchronous calls but there is no error handling in case these operations fail. Also, the `console.warn(e)` statement will only log the error to the console and won't provide any feedback to the user. Consider adding error handling logic that provides feedback to the user when an error occurs.
```diff
  const onPressDM = async () => {
    try {
      setLoading({...loading, loadingDm: true});
      if (!item?.anon_user_info_emoji_name) {
        const channelName = [username, profile?.myProfile?.username].join(',');
        const selectedUser = {
          user: {
            name: channelName,
            image: item?.actor?.data?.profile_pic_url || DEFAULT_PROFILE_PIC_PATH
          }
        };
        const members = [item?.actor?.id, profile.myProfile.user_id];
        await createSignChat(members, selectedUser);
      } else {
        await sendMessageDM(item.id, 'post', 'SIGNED');
      }
    } catch (e) {
-       console.warn(e);
+       // Handle error and provide feedback to the user
    } finally {
      refSheet.current.close();
      setLoading({...loading, loadingDm: false});
    }
  };

  const onPressDMAnon = async () => {
    try {
      setLoading({...loading, loadingDmAnon: true});
      await sendMessageDM(item.id, 'post', 'ANONYMOUS');
    } catch (e) {
-       console.warn(e);
+       // Handle error and provide feedback to the user
    } finally {
      refSheet.current.close();
      setLoading({...loading, loadingDmAnon: false});
    }
  };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->